scripts/analysisTools/w_mass_13TeV/run2Dsmoothing.py

```diff
@@ -689,23 +698,54 @@ def runSmoothing(inputfile, histname, outdir, step, args, effHist=None):
     # efficiencies made with scripts/analysisTools/w_mass_13TeV/makeWMCefficiency3D.py
     #
     # sfFolder = data_dir + "/muonSF/"
-    # effSmoothFile = f"{sfFolder}efficiencies3D_rebinUt2.pkl.lz4"
-    sfFolder = data_dir + "/muonSF/"
-    sfFolderVtxAgn = data_dir + "/muonSF/intermediate_vtxAgnosticIso/"
-    effSmoothFile = sfFolderVtxAgn + "efficiencies3D_rebinUt2_vtxAgnPfRelIso04.pkl.lz4"
-    #
-    inputRootFile = {
-        "iso": f"{sfFolderVtxAgn}iso3DSFVQTsingularity.root",
-        "isonotrig": f"{sfFolderVtxAgn}isonotrig3DSFVQTsingularity.root",
-        "isoantitrig": f"{sfFolderVtxAgn}isofailtrig3DSFVQTsingularity.root",
-        "triggerplus": f"{sfFolder}triggerplus3DSFVQTextended.root",
-        "triggerminus": f"{sfFolder}triggerminus3DSFVQTextended.root",
+    
+    default_inputRootFiles = {}
+    default_inputRootFiles["2016PostVFP"] = {
+        "iso": data_dir + "/muonSF/intermediate_vtxAgnosticIso/iso3DSFVQTsingularity.root",
+        "isonotrig": data_dir + "/muonSF/intermediate_vtxAgnosticIso/isonotrig3DSFVQTsingularity.root",
+        "isoantitrig": data_dir + "/muonSF/intermediate_vtxAgnosticIso/isofailtrig3DSFVQTsingularity.root",
+        "triggerplus": data_dir + "/muonSF/triggerplus3DSFVQTextended.root",
+        "triggerminus": data_dir + "/muonSF/triggerminus3DSFVQTextended.root",
+    }
+
+    for yr in ["2017", "2018"]:
+        default_inputRootFiles[yr] = {
+            "iso": data_dir + f"/muonSF/{yr}/3dSF/3dsfisolation.root",
+            "isonotrig": data_dir + f"/muonSF/{yr}/3dSF/3dsfisonotrig.root",
+            "triggerplus": data_dir + f"/muonSF/{yr}/3dSF/3dsftriggerplus.root",
+            "triggerminus": data_dir + f"/muonSF/{yr}/3dSF//3dsftriggerminus.root",
+        }
+
+    histNames_newEras = {
+        "iso": "Isolation",
+        "isonotrig": "IsolationNoTrigger",
+        "isoantitrig": "IsolationAntiTrigger",
+        "triggerplus": "TriggerPlus",
+        "triggerminus": "TriggerMinus",
     }
 
     parser = common_plot_parser()
+    parser.add_argument(
+        "-i",
+        "--indir",
+        default = None,
+        help="Input directory where the 3d efficiencies measured on data are stored"
+    )
     parser.add_argument(
         "outdir", type=str, nargs=1, help="output directory to save things"
     )
+    parser.add_argument(
+        "--era", 
+        type=str,
+        default="2016PostVFP",
+        choices=["2016PostVFP", "2017", "2018"]
+    )
+    parser.add_argument(
+        "--eff_filename",
+        type=str,
+        default="efficiencies3D.pkl.lz4",
```

Comment:
Since the baseline is 2016PostVFP, and the files are already selected within the code, I would use None as the default of this option, and use this default you have here within the "if" scope later in the code. Otherwise if a new user sees the code he would think this is the actual default regardless of other things, while in fact it is only for 2017/2018 at the moment.
Then if the user needs a totally different file they can just use this option. In this case this option should also allow changing the default files of 2016.